source/common/network/upstream_server_name.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include "envoy/stream_info/filter_state.h"
+
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Network {
+
+/**
+ * Server name to set in the upstream connection. The filters like tcp_proxy should use this
```

Comment:
Was there a reason we couldn't get SNI from the existing request info available to filters? I think in general this is the right approach as we grow the number of things we're passing down, just I would have thought we already had SNI available in RequestInfo.